util/dvsim/SimCfg.py

```diff
@@ -646,40 +645,51 @@ def create_failure_message(test, line, context):
         return results_str
 
     def gen_results_summary(self):
+        '''Generate the summary results table.
 
-        # sim summary result has 5 columns from each SimCfg.results_summary
-        header = ["Name", "Passing", "Total", "Pass Rate"]
-        if self.cov_report_deploy is not None:
-            header.append('Coverage')
-        table = []
-        colalign = ("center", ) * len(header)
-        for item in self.cfgs:
-            row = []
-            for title in item.results_summary:
-                row.append(item.results_summary[title])
-            if row:
-                table.append(row)
-        self.results_summary_md = "## " + self.results_title + " (Summary)\n"
-        self.results_summary_md += "### " + self.timestamp_long + "\n"
+        This method is specific to the primary cfg. It summarizes the results
+        from each individual cfg in a markdown table.
+
+        Prints the generated summary markdown text to stdout and returns it.
+        '''
+
+        text = [f"## {self.results_title} (Summary)"]
+        text += [f"### {self.timestamp_long}"]
         if self.revision:
-            self.results_summary_md += "### " + self.revision + "\n"
-        self.results_summary_md += "### Branch: " + self.branch + "\n"
-        if table:
-            self.results_summary_md += tabulate(table,
-                                                headers=header,
-                                                tablefmt="pipe",
-                                                colalign=colalign)
-        else:
-            self.results_summary_md += "\nNo results to display.\n"
+            text += [f"### {self.revision}"]
+        text += [f"### Branch: {self.branch}"]
 
+        if not self.cfgs:
+            # The blanks before and after the table get converted to \n's.
+            text += ["", "No results to display.", ""]
+
+        else:
+            table = []
+            # Pick the header from the first cfg and ensure the rest match.
```

Comment:
Can you extend this comment to explain that we do the "ensuring" part when looping over cfgs below? I thought this check was missing when I read it at first.

Reply:
Done.